src/update_collection_metadata.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+
+import github3
+from prefect import flow, task
+from prefect.blocks.system import Secret
+from prefect.deployments import run_deployment
+from prefect.states import Completed
+
+from utils import LatestReleases
+
+
+@task
+async def collection_needs_update(collection_name: str) -> bool:
+    """
+    Checks if the collection needs to be updated.
+    """
+    github_token = await Secret.load("github-token")
+    gh = github3.login(token=github_token.get())
+
+    collection_repo = gh.repository("PrefectHQ", collection_name)
+
+    latest_releases = await LatestReleases.load("collections")
+    latest_observed_release = latest_releases.get(collection_name)
+    latest_release = collection_repo.latest_release().tag_name
```

Comment:
I wonder if we can drop the token / block requirement so there's less setup required for whoever will be using this in the future.

For latest release, I think we can simply: `httpx.get("https://api.github.com/repos/PrefectHQ/prefect-aws/tags").json()`, sort by name, and extract the latest.

For the latest release block, I wonder if we can introspect the repo's directory to locate what the latest release is and compare?

Reply:
On second thought, maybe we do need the GH token to submit a PR. If so, maybe needs documentation on what permissions the token needs (if not done so already).

Reply:
yep I will need to figure out minimal permissions here